### PR TITLE
s3: add SourceBucketLocation for CrossLocationLoggingProhibitions exception

### PR DESCRIPTION
**Pull request by sannya-singal**
## Motivation
While fixing a test case: `test_put_bucket_logging_wrong_target` for multiple accounts and regions in #9713, found AWS parity issues with parameters of `CrossLocationLoggingProhibitted` exception. 

## Changes
This PR: 
- updates the exception `CrossLocationLoggingProhibitted` with `SourceBucketLocation`.
- update the raised exceptions in the `PutBucketLogging` handler. 
- validates the change with `test_put_bucket_logging_cross_locations` test case.

### CODE REVIEW

File: localstack/services/s3/v3/provider.py [hunk 1]
```diff
@@ -3283,10 +3284,15 @@ def put_bucket_logging(
                 TargetBucket=target_bucket_name,
             )
 
-        if target_s3_bucket.bucket_region != s3_bucket.bucket_region:
+        source_bucket_region = s3_bucket.bucket_region
+        if target_s3_bucket.bucket_region != source_bucket_region:
             raise CrossLocationLoggingProhibitted(
                 "Cross S3 location logging not allowed. ",
                 TargetBucketLocation=target_s3_bucket.bucket_region,
+            ) if source_bucket_region == AWS_REGION_US_EAST_1 else CrossLocationLoggingProhibitted(
```
Comment (bentsku) [inline localstack/services/s3/v3/provider.py:3292]: I had never seen a `Raise X if something else Y`! That's pretty cool

**Review by bentsku — APPROVED** [review]
This is a great catch, and a great fix! Nice use of the patches, great test, honestly this is great 💯!! Thank you so much for fixing this. 🙏 everything LGTM!

So now that I see this, I'll also approve #9713, as having `us-east-1` is mandatory for the snapshot tests to work, so need to update the PR, it's good how it is? Thanks for checking it and validating it!! Great work

**Comment by sannya-singal** [comment]
`localstact-ext` Integration tests pipeline run here: https://github.com/localstack/localstack-ext/actions/runs/7004532749

**Comment by sannya-singal** [comment]
> This is a great catch, and a great fix! Nice use of the patches, great test, honestly this is great 💯!! Thank you so much for fixing this. 🙏 everything LGTM!
> 
> So now that I see this, I'll also approve #9713, as having `us-east-1` is mandatory for the snapshot tests to work, so need to update the PR, it's good how it is? Thanks for checking it and validating it!! Great work

Thank you so much for the review @bentsku 🙇 🙏 🚀

**Pull request merged**